js/ui/windowManager.js

```diff
@@ -2140,66 +2179,51 @@ const CodingManager = new Lang.Class({
         let focusedWindow = global.display.get_focus_window();
         if (!focusedWindow)
             return;
-        if (!session.windowApp.get_meta_window())
-            return;
 
-        if (session.buttonApp)
-            session.buttonApp.hide();
-        if (session.buttonBuilder)
-            session.buttonBuilder.hide();
+        let previousFocused = this._previousFocusedWindow;
+        this._previousFocusedWindow = focusedWindow;
 
-        if (focusedWindow.get_stable_sequence() === session.windowApp.get_meta_window().get_stable_sequence()){
-            if (session.state === SessionState.APP) {
-                session.buttonApp.show();
-                return;
+        session.buttonApp.hide();
+        if (focusedWindow === session.windowApp.get_meta_window()) {
+            if (session.windowBuilder)
+            {
+                if (session.windowBuilder.get_meta_window() === previousFocused) {
+                    this._rotateInActors.push(session.windowApp);
+                    this._rotateOutActors.push(session.windowBuilder);
+                    this._animate(session.windowBuilder,
+                                  session.windowApp,
+                                  Gtk.DirectionType.RIGHT);
+                    session.buttonBuilder.hide();
+                    session.buttonApp.show();
+                    session.state = SessionState.APP;
```

Comment:
Just to clarify here - are we doing a builder -> app transition here when the builder window loses focus?